src/context/DataContext/DataReducer.js

```diff
@@ -0,0 +1,66 @@
+export const initialState = {
+  videos: [],
+  categories: [],
+  watchLater: [],
+  likedVideos: [],
+};
+
+export const filterReducer = (state, action) => {
+  switch (action.type) {
+    case "SORT":
+      return {
+        ...state,
+        sort: action.payload,
+      };
+
+    case "CATEGORIES":
+      return {
+        ...state,
+        category: action.payload.isDel ? "" : action.payload.categoryName,
```

Comment:
what is  isDel ??

Reply:
A boolean attribute that denotes if I wanted to remove the current category or add it.

If the category is already selected and user clicked on it again, then `isDel` will be true and category filter will be resetted to `All`